x/denommetadata/types/hooks.go

```diff
@@ -0,0 +1,36 @@
+package types
+
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// DenomMetadataHooks event hooks for denom metadata creation/update
+type DenomMetadataHooks interface {
```

Comment:
@omritoptix u think we need to seperate the cases?
IMO it can be single hook called when metadata is set

Reply:
@mtsitrin we could, but I think it's cleaner to have this separation as @VictorTrustyDev mentioned that ERC20 updating metadata after creation is not really taking effect (also possibly for future use cases).

in the hooks registered on the hub for VFC we only listen to onCreate (on update we do nothing) just updating the denometada on the hub but not the contract. 

Not sure how it will be used later on but as we opted no to update the contract onUpdate than I think it's better to leave it as is currently.